### PR TITLE
Pass props on to containers in remaining components

### PR DESCRIPTION
**Pull request by gavinorland**
This PR takes care of https://github.com/stevesims/govuk-frederic/issues/49

### CODE REVIEW

File: components/countdown-text-area/src/index.js [hunk 1]
```diff
@@ -86,7 +86,7 @@ const CountdownTextArea = (props) => {
   const over = showRemaining && remaining <= 0;
 
   return (
-    <StyledTextArea over={over}>
+    <StyledTextArea over={over} {...props}>
```
Comment (stevesims) [inline components/countdown-text-area/src/index.js]: i'm not sure this is right

feels like for here that we may wish to extract and set `id` and `className` only - the other props are only relevant for the `StyledTextAreaField` component

alternatively we just leave this as-is - i'm not sure of the advantage of setting any props on the container here

File: components/header-button/src/index.js [hunk 2]
```diff
@@ -84,8 +84,8 @@ const Button = styled('button')(
  * <HeaderButton disabled>Off</HeaderButton>
  * ```
  */
-const HeaderButton = ({ active, children, disabled, ...props }) => (
-  <Button {...props} active={active} disabled={disabled}>
+const HeaderButton = ({ children, ...props }) => (
```
Comment (stevesims) [inline components/header-button/src/index.js]: this component can just become
```
const HeaderButton = props => <Button {...props} />;
```

File: components/keyline/src/index.js [hunk 3]
```diff
@@ -37,8 +37,8 @@ const KeylineWrapper = styled('span')(({ inlineBlock, borderColor }) => ({
  * </Keyline>
  * ```
  */
-const Keyline = ({ children, inlineBlock, borderColor }) => (
-  <KeylineWrapper inlineBlock={inlineBlock} borderColor={borderColor}>
+const Keyline = ({ children, ...props }) => (
```
Comment (stevesims) [inline components/keyline/src/index.js]: Similarly this component can become
```
const Keyline = props => <KeylineWrapper {...props} />;
```

File: components/countdown-text-area/src/index.js (hunk 1 above)
Comment (gavinorland) [reply components/countdown-text-area/src/index.js]: Yes, would mainly just be to handle props like `className` passed down from HOCs. Maybe we should leave this as is for now, now it's done, but can can reconsider later.

File: components/header-button/src/index.js (hunk 2 above)
Comment (gavinorland) [reply components/header-button/src/index.js]: Nice 👍

File: components/countdown-text-area/src/index.js (hunk 1 above)
Comment (stevesims) [reply components/countdown-text-area/src/index.js]: by "as-is" i meant before this edit was made :grin:

we can leave this in, as passing down extraneous properties isn't necessarily a biggie, but i'd be more comfortable if we at least put a note against this in the code to reconsider what we're doing here

Comment (gavinorland) [reply components/countdown-text-area/src/index.js]: Yes, I know what you mean - we can't explicitly see what is being sent down. I'll raise an issue, so it's logged, for us to consider extracting and sending down only specific props in the future.

**Pull request merged**